modeshape-jcr/src/main/java/org/modeshape/jcr/RepositoryLockManager.java

```diff
@@ -131,14 +132,21 @@ protected void refreshFromSystem() {
                     NodeKey lockKey = ref.getKey();
                     logger.warn(JcrI18n.lockNotFound, lockKey);
                     mutableLocks.removeChild(systemCache, lockKey);
+                    corruptedLocks.add(lockKey);
                     continue;
                 }
-                
                 ModeShapeLock lock = new ModeShapeLock(node, systemCache);
                 locksByNodeKey.put(lock.getLockedNodeKey(), lock);
             }
             if (mutableLocks != null) {
                 system.save();
+                for (Iterator<Map.Entry<NodeKey, ModeShapeLock>> locksIterator = locksByNodeKey.entrySet().iterator(); 
+                     locksIterator.hasNext();) {
+                    NodeKey lockKey = locksIterator.next().getValue().getLockKey();
+                    if (corruptedLocks.contains(lockKey)) {
+                        locksIterator.remove();
+                    }
+                }
```

Comment:
yes, its purpose is to clear any potential in-memory locks stored in the `locksByNodeKey` member.